code/modules/power/cell.dm

```diff
@@ -17,14 +17,26 @@
 	var/rigged = 0		// true if rigged to explode
 	var/minor_fault = 0 //If not 100% reliable, it will build up faults.
 	var/chargerate = 100 //how much power is given every tick in a recharger
+	var/self_recharge = 0 //does it self recharge, over time, or not?

```

Comment:
You should do what energy guns do with on_varedit() so admins can edit this var to have a cell recharge.
https://github.com/tgstation/-tg-station/blob/master/code/modules/projectiles/guns/energy.dm#L160